sudo/src/pipeline.rs

```diff
@@ -0,0 +1,105 @@
+use std::{process::exit, sync::atomic::Ordering};
+
+use sudo_cli::SudoOptions;
+use sudo_common::{Context, Error};
+use sudo_env::environment;
+use sudo_exec::ExitReason;
+use sudoers::{Authorization, DirChange, Policy, PreJudgementPolicy};
+
+pub trait PolicyPlugin {
+    type PreJudgementPolicy: PreJudgementPolicy;
```

Comment:
I.e. (continuing the theme above), the above line can go and we can just replace `PreJudgementPolicy` with `PolicyPlugin`. (Note that "PreJudgementPolicy" isn't my favorite name anyway and PolicyPlugin is a much better name).